### PR TITLE
L10n packaging

### PR DESCRIPTION
**Pull request by chrisgarrity**
Moved towards being able to publish this as a package.

Added linting for json to ensure that downloads from Transifex are not broken.

Not yet done:
- Tests
- Generating localization for components other than gui

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+  "name": "scratch-l10n",
+  "version": "0.1.0",
+  "description": "Localization for the Scratch 3.0 components",
+  "main": "./dist/l10n.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "build:data": "babel-node scripts/build-data",
+    "build": "npm run clean && npm run build:data && webpack --progress --colors --bail",
+    "clean": "rimraf ./dist ./locales && mkdirp dist locales",
+    "lint": "eslint . && jlint --silent --glob './gui/*.json'",
```
Comment (rschamp) [inline package.json]: In www we do json linting via eslint and a plugin — I think that would be simpler than a separate linter for json.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const defaultsDeep = require('lodash.defaultsdeep');
+const path = require('path');
+const webpack = require('webpack');
+
+
+const base = {
+    devtool: 'cheap-module-source-map',
+    module: {
+        rules: [{
+            test: /\.jsx?$/,
```
Comment (rschamp) [inline webpack.config.js]: I don't think we'll ever need jsx, will we?

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            })
+        ] : [])
+};
+
+module.exports = [
+    // For use as a library
+    defaultsDeep({}, base, {
+        externals: {
+            'react': 'react',
+            'react-dom': 'react-dom'
```
Comment (rschamp) [inline webpack.config.js]: Nor React/ReactDOM?

File: package.json [hunk 4]
```diff
@@ -1,19 +1,43 @@
...
+    "babel-preset-es2015": "^6.24.1",
+    "eslint": "^4.6.1",
+    "eslint-config-scratch": "^4.0.0",
+    "eslint-plugin-import": "^2.7.0",
+    "eslint-plugin-react": "^7.2.1",
+    "jlint": "^3.0.2",
+    "lodash.defaultsdeep": "4.6.0",
+    "mkdirp": "^0.5.1",
+    "prop-types": "^15.5.10",
+    "react": "15.6.1",
```
Comment (rschamp) [inline package.json]: I think we can omit `prop-types` and `react`.

File: package.json [hunk 5]
```diff
@@ -1,19 +1,43 @@
...
+    "babel-cli": "^6.26.0",
+    "babel-core": "^6.26.0",
+    "babel-eslint": "^7.2.3",
+    "babel-loader": "^7.1.2",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
+    "babel-preset-es2015": "^6.24.1",
+    "eslint": "^4.6.1",
+    "eslint-config-scratch": "^4.0.0",
+    "eslint-plugin-import": "^2.7.0",
+    "eslint-plugin-react": "^7.2.1",
```
Comment (rschamp) [inline package.json]: We can omit `eslint-plugin-react` too I think.

File: gui/ar.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    "targetPane.addSprite": {
+        "message": "أَضِف كائن",
+        "description": "Button to add a sprite in the target pane"
+    },
+    "soundEditor.louder": {
+        "message": "أعلى",
+        "description": "Title of the button to apply the louder effect"
+    }
+}
```
[the remark this answers is not in the document]
Comment (chrisgarrity) [reply gui/ar.json]: I'll check - these are what's downloaded from Transifex, so it depends on whether I can configure their output.

File: package.json (hunk 1 above)
Comment (chrisgarrity) [reply package.json]: Definitely!

File: webpack.config.js (hunk 2 above)
Comment (chrisgarrity) [reply webpack.config.js]: The rule checks for `.js` or `.jsx`. We do need .js, so I can change the rule, I just left it as-is for consistency with other scratch repos

File: gui/ar.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (chrisgarrity) [reply gui/ar.json]: Doesn't look like there's any way to force Transifex to generate with a newline. Hopefully these will always be generated the same way by Transifex so it won't become an issue.

File: .babelrc [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "plugins": [
+        "transform-object-rest-spread"
+    ],
+    "presets": ["es2015"]
+}
```
[the remark this answers is not in the document]
Comment (chrisgarrity) [reply .babelrc]: Not sure why this isn't showing as outdated, but the newline is added!

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+            test: /\.js$/,
+            loader: 'babel-loader',
+            include: path.resolve(__dirname, 'src')
+        }]
+    },
+    plugins: []
+        .concat(process.env.NODE_ENV === 'production' ? [
+            new webpack.optimize.UglifyJsPlugin({
+                include: /\.min\.js$/,
+                minimize: true
```
Comment (rschamp) [inline webpack.config.js]: Looking at it again, I don't think we need this — we won't be using minified js from this repo.

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
+const defaultsDeep = require('lodash.defaultsdeep');
+const path = require('path');
+const webpack = require('webpack');
+
+
+const base = {
```
Comment (rschamp) [inline webpack.config.js]: Also, since we're only building one thing, this can be collapsed into a single export (no base that gets extended).

**Pull request merged**